src/app/InteractionModelEngine.cpp

```diff
@@ -688,10 +706,8 @@ bool InteractionModelEngine::EnsureResourceForSubscription(FabricIndex aFabricIn
     }
 
     const auto evictAndUpdateResourceUsage = [&](FabricIndex fabricIndex, bool forceEvict) {
-        bool ret           = TrimFabric(fabricIndex, forceEvict);
-        usedAttributePaths = mAttributePathPool.Allocated();
-        usedEventPaths     = mEventPathPool.Allocated();
-        usedReadHandlers   = mReadHandlers.Allocated();
+        bool ret = TrimFabric(fabricIndex, forceEvict);
+        countResourceUsage();
```

Comment:
This looks great, but do we still need the "Reserve the last ReadHandler for ReadInteraction" bits in `InteractionModelEngine::OnReadInitialRequest`?  It seems to me that this function ensures that subscriptions never use up more than (total read handlers - MAX_FABRICS), so that block of code is redundant (and not really right, since it's not reserving one per fabric anyway).  Or am I missing something?

Reply:
We no longer need that check, and I should remove it in previous subscription quota PR.

Removed that check.